tools/gm-to-obj/gm-to-obj.py

```diff
@@ -0,0 +1,298 @@
+                ('group_name', c_long),
+                ('name', c_long),
+                ('r', c_float),
+                ('g', c_float),
+                ('b', c_float),
+                ('range', c_float),
+                ('pos', c_vector),
+                ('atten', c_float*3),
+                ('inner', c_float),
+                ('outer', c_float),
+                ('falloff', c_float),
+                ('dir', c_vector),
+                ]
+
+
+class rdf_label(Structure):
+    _fields_ = [('group_name', c_long),
+                ('name', c_long),
+                ('flags', c_long),
+                ('m', c_float*16),
+                ('bones', c_long*4),
+                ('weight', c_float*4)]
+
+
+class rdf_object(Structure):
+    _fields_ = [('group_name', c_long),
+                ('name', c_long),
+                ('flags', c_long),
+                ('center', c_vector),
+                ('radius', c_float),
+                ('vertex_buff', c_long),
+                ('ntriangles', c_long),
+                ('striangle', c_long),
+                ('nvertices', c_long),
+                ('svertex', c_long),
+                ('material', c_long),
+                ('lights', c_long*8),
+                ('bones', c_long*4),
+                ('atriangles', c_long)]
+
+
+class rdf_triangle(Structure):
+    _fields_ = [
+        ('vindex', c_ushort*3)
+    ]
+
+
+class rdf_vertexbuff(Structure):
+    _fields_ = [
+        ('type', c_long),
+        ('size', c_long)
+    ]
+
+
+class rdf_vertex0(Structure):
+    _fields_ = [
+        ('pos', c_vector),
+        ('norm', c_vector),
+        ('color', c_long),
+        ('tu0', c_float),
+        ('tv0', c_float)
+    ]
+
+
+class rdf_bsphead(Structure):
+    _fields_ = [
+        ('nnodes', c_long),
+        ('nvertices', c_long),
+        ('ntriangles', c_long),
+    ]
+
+
+class bsp_node(Structure):
+    _fields_ = [
+        ('norm', c_vector),
+        ('pd', c_long),
+        ('node', c_ulong),
+        ('sign', c_ulong),
+        ('left', c_ulong),
+        ('nfaces', c_ulong),
+        ('right', c_ulong),
+        ('type', c_ulong),
+        ('face', c_long),
+    ]
+
+
+class rdf_bsptriangle(Structure):
+    _fields_ = [
+        ('vindex', c_char*9)
+    ]
+
+
+class vertex_buffer(Structure):
+    _fields_ = [
+        ('type', c_ulong),
+        ('size', c_long),
+        ('stride', c_ulong),
+        ('nverts', c_ulong),
+        ('dev_buff', c_ulong)
+    ]
+
+
+def printArray(arr):
+    # Print all properties of an arr/obj
+    for o in arr:
+        print("___________________")
+        for field_name, field_type in o._fields_:
+            print(field_name, getattr(o, field_name))
+
+
+def getArrayof(struct, size):
+    # Read an array of the struct type with the fiven size
+    res = []
+    for _ in range(0, size):
+        m = eval(struct)()
+        f.readinto(m)
+        res.append(m)
+    return res
+
+
+user_input = input("Enter path of the .gm file ")
```

Comment:
AFAIK, it's considered a good practice to wrap the code in `if __name__ == '__main__':`

Reply:
By adding that, the script cannot be called by other scripts. Are you sure you want that?

Reply:
I believe, argument parsing should be wrapped in `if __name__ == '__main__':` and the actual logic of a script should be in a function to allow calling from other scripts, as you said